java/vector/src/main/java/org/apache/arrow/vector/compare/RangeEqualsVisitor.java

```diff
@@ -183,21 +204,14 @@ protected boolean compareBaseVariableWidthVectors(BaseVariableWidthVector left)
         return false;
       }
 
-      int offsetWidth = BaseVariableWidthVector.OFFSET_WIDTH;
-
-      if (!isNull) {
```

Comment:
Should we keep this branch since there's no need to compare value length if isNull？

Reply:
Yes I think so, We should probably have a unit test that fails, covers this case as well.

Reply:
Actually maybe not for variable width fields.